typing/typedecl_variance.ml

```diff
@@ -322,11 +322,12 @@ let compute_variance_decl env ~check decl (required, _ as rloc) =
     Option.map (fun id -> Type_declaration (id, decl)) check
   in
   let abstract = Btype.type_kind_is_abstract decl in
-  if (abstract || decl.type_kind = Type_open) && decl.type_manifest = None then
+  match decl with
+  | {type_kind = (Type_abstract _ | Type_open); type_manifest = None} ->
     List.map
       (fun (c, n, i) -> make (not n) (not c) (not abstract || i))
       required
-  else begin
+  | _ -> (* TODO: list every type_kind *)
```

Comment:
Do you plan to implement this TODO in this PR or a later one?

Reply:
Let me do it in this PR.